playwright/utils/auth.ts

```diff
@@ -0,0 +1,71 @@
+import { BasicApiService } from "@api/basics";
+import { USER_PERMISSION, UserPermission } from "@data/userPermissions";
+import { request } from "@playwright/test";
+import fs from "fs";
+import path from "path";
+
+/**
+ * Retrieves the storage state for a given user permission.
+ * If the storage state does not exist, it creates the necessary directories,
+ * logs in the user via API, and saves the storage state to a file.
+ *
+ * @param {UserPermission} permission - The user permission for which to retrieve the storage state.
+ * @returns {Promise<string>} - A promise that resolves to the path of the storage state file.
+ */
+export const getStorageState = async (permission: UserPermission | "admin"): Promise<string> => {
+  const tempDir = path.join(__dirname, "../.auth");
+  const storageStatePath = path.join(tempDir, `${permission}.json`);
+
+  // Create the .auth directory if it does not exist.
+  if (!fs.existsSync(tempDir)) {
+    fs.mkdirSync(tempDir, { recursive: true });
+  }
+
+  // If the storage state does not exist, log in the user via API and save the storage state to a file.
+  if (!fs.existsSync(storageStatePath)) {
+    const apiRequestContext = await request.newContext({
+      baseURL: process.env.BASE_URL!,
+    });
+
+    const basicApiService = new BasicApiService(apiRequestContext);
+
+    const email = getEmailForPermission(permission);
+    const password = getPasswordForPermission(permission);
+
+    await basicApiService.logInUserViaApi({ email, password });
+
+    const loginJsonInfo = await apiRequestContext.storageState();
+
+    loginJsonInfo.origins.push({
+      origin: process.env.BASE_URL!,
+      localStorage: [
+        {
+          name: "_saleorRefreshToken",
+          value: loginJsonInfo.cookies[0].value,
```

Comment:
Since this line of code fails very often I would check if there's an error message in `loginJsonInfo` and throw it before so that it's easier to debug later